magento/pwa-studio
# PWA-1674: View Product Attributes values on PDP - Text input, Multi-s…

*In scope:* Multi-select, Boolean, Date
As a Merchant I want to show product attributes values on PWA Storefront using native capabilities *so that* I manage product information without ongoing developer help



## Description

*In scope:* Multi-select, Boolean, Date
As a Merchant I want to show product attributes values on PWA Storefront using native capabilities *so that* I manage product information without ongoing developer help

## Related Issue







Closes PWA-1674

## Acceptance




*Acceptance Criteria*
 1. User navigates to the product detail page and sees product attributes to be displayed in the _Details_ section of product information based on rank set in Store>Product>Attribute>Storefront Properties>Position. If the rank is not set to define position based on it, product attributes sorted alphabetically.

1. User sees attributes displayed with a label corresponding to a store view

1. User sees the date value in the localized vesting according to the store view setting 

### Verification Stakeholders



### Specification




## Verification Steps





*Preconditions*
 1. Web store with 2 store views created
1. Product attributes with input types Multi-select, Boolean and Date created in Admin>Store>Attributes>Product
1. Product attributes are enabled for Product Detail Page in Admin>Store>Attributes> Visible on Catalog Pages on Storefront
1. Product attributes have labels localized for each store view in Admin>Store>Attributes>Manage Labels
1. Products that use product attributes above are created in Admin Panel
1. User is browsing  products mentioned above on Venia PWA Storefront


#### Test scenario(s) for direct fix/feature





*Test Plan*
1. Setup data in backend and verify - 
 ** Product attributes Multi-select, Boolean, Date are displayed for both simple and configurable products. (Virtual products also if that is delivered when this ticket hits board)
 ** Verify the display order is based on rank and if rank is not set then its alphabetical order
 ** Verify above scenario's for store view 2
 ** Verify localized label displayed per backend config.
1. Verify in different browser special date attribute.

*Cypress test coverage*
1. Add cypress integration mock data test to assert attributes display. (Update test if already exists)
 

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
